packages/product-json-to-csv/src/map-product-data.js

```diff
@@ -74,92 +106,257 @@ export default class ProductMapping {
+    })
+
+    // add labels in format "attribName.en: labelEnValue" to all attributes
+    return _.merge(mappedAttribute, labels)
+  }
+
+  _mapLabelsToAllAttributes(name: string, labels: Array<Object>): Object {
+    const mappedValues: Object = {}
+    const languages: Array<string> = _(labels)
+      .map(Object.keys)
+      .flatten()
+      .uniq()
+      .value()
+
+    languages.forEach(lang => {
+      const headerKey: string = `${name}.${lang}`
+      const values = []
+
+      labels.forEach(label => {
+        values.push(label[lang] || '')
+      })
+
+      mappedValues[headerKey] = values.join(this.multiValDel)
+    })
+
+    return mappedValues
+  }
+
+  _mapSetAttribute(name: string, values: Array<any>) {
+    const [firstValue] = values
+    let mappedValue: string = ''
+
+    // empty set
+    if (values.length === 0) mappedValue = ''
+
+    // string, boolean, number
+    if (!_.isObject(firstValue)) mappedValue = values.join(this.multiValDel)
+    else if (firstValue.id && firstValue.typeId)
+      // reference
+      mappedValue = values
+        .map(value => ProductMapping._mapReference(value))
+        .join(this.multiValDel)
+    else if (firstValue.key && !_.isObject(firstValue.label))
+      // enum
+      mappedValue = _.map(values, 'key').join(this.multiValDel)
+    else if (firstValue.key && _.isObject(firstValue.label)) {
+      // lenum
+      const labels = _.map(values, 'label')
+      // map all language labels
+      const mappedValues = this._mapLabelsToAllAttributes(name, labels)
+      // add lenum keys as a main attribute value
+      mappedValues[name] = _.map(values, 'key').join(this.multiValDel)
+      return mappedValues
+    } else {
+      // ltext
+      const mappedValues = this._mapLabelsToAllAttributes(name, values)
+      // copy value from selected language as a main attribute value
+      mappedValues[name] = mappedValues[`${name}.${this.lang}`]
+      return mappedValues
+    }
+
+    return {
+      [name]: mappedValue,
+    }
+  }
+
+  /**
+   * Method will take an attribute and map it to string
+   * @param attribute Attribute which should be mapped
+   * @param allAttributes Result object with all attributes
+   * @returns {*} Mapped attribute
+   * @private
+   */
+  _mapAttribute(attribute: Attribute): Object {
+    const { name, value } = attribute
+    let mappedAttribute: Object = {}
+
+    if (_.isObject(value) && !_.isArray(value)) {
+      // ltext, enum, lenum
+      if (value.id && value.typeId)
+        // reference
+        mappedAttribute[name] = ProductMapping._mapReference(value)
+      else if (value.key && !_.isUndefined(value.label))
+        // ENUM or LENUM attribute
+        mappedAttribute = ProductMapping._mapLenumOrEnumAttribute(name, value)
+      else
+        // LTEXT attribute
+        mappedAttribute = this._mapLtextAttribute(name, value)
+    } else if (_.isArray(value)) {
+      // SET attribute
+      mappedAttribute = this._mapSetAttribute(name, value)
```

Comment:
sorry, but isn't this the same as 

```js
function mapAttribute(attribute) {
 if (Array.isArray(attribute.value)) {
   const mappedValues = attribute.value.map((value) => ({ value }))
 }
}
```

Reply:
Hi @adnasa .. I am not sure if this will work - for example for `set of ltext` you have values:
```js
{
	name: 'attributeName',
	value: [
		{
			en: 'abcd',
			de: 'dcba'
		},
		{
			en: '123', 
			de: '456' 
		}
	]
}
```

And you want to get:
```js
{
	'attributeName': 'abcd,123', // based on selected language
	'attributeName.en': 'abcd,123',
	'attributeName.de': 'dcba,456'
}
```

In your case you would get:
```js
[
	{
		'attributeName': 'abcd', // based on selected language
		'attributeName.en': 'abcd',
		'attributeName.de': 'dcba'
	},
	{
		'attributeName': '123', // based on selected language
		'attributeName.en': '123',
		'attributeName.de': '456'
	}
]
```

I can implement it like this but then we would have to loop through all items in resulting array and merge them, WDYT?